storm-client/src/jvm/org/apache/storm/utils/NimbusClient.java

```diff
@@ -126,7 +137,9 @@ public static NimbusClient getConfiguredClientAs(Map<String, Object> conf, Strin
                 nimbusSummary = client.getClient().getLeader();
                 if (nimbusSummary != null) {
                     String leaderNimbus = nimbusSummary.get_host() + ":" + nimbusSummary.get_port();
-                    LOG.info("Found leader nimbus : {}", leaderNimbus);
+                    if (LOG.isDebugEnabled() || shouldLogLeader(leaderNimbus)) {
```

Comment:
nit: There's mismatch between checked log level and actual log level of message. I think `shouldLogLeader` will reduce verbose of message so personally I prefer to leave message whenever shouldLogLeader is true.

Reply:
I will adjust it so it is clearer.